src/Action/ExceptionAction.php

```diff
@@ -80,6 +83,12 @@ public function __invoke(FlattenException $exception, Request $request): Respons
         $headers['X-Content-Type-Options'] = 'nosniff';
         $headers['X-Frame-Options'] = 'deny';
 
-        return new Response($this->serializer->serialize($exception, $format['key']), $exception->getStatusCode(), $headers);
+        $response = new Response(null, $exception->getStatusCode(), $headers);
+
+        if (!$this->debug && $response->isServerError()) {
+            $exception->setMessage(Response::$statusTexts[$response->getStatusCode()]);
```

Comment:
I thought the same thing but currently the `FlattenException` is already altered ([ExceptionAction.php#L71](https://github.com/api-platform/core/blob/master/src/Action/ExceptionAction.php#L71)). But in the same time, the original exception is not altered because `FlattenException` is just _a wrap to be able to serialize it_. The original one is logged and collected by the profiler before the request is duplicated and the `FlattenException` is created.